homeassistant/components/filesize/config_flow.py

```diff
@@ -21,14 +22,15 @@
 
 def validate_path(hass: HomeAssistant, path: str) -> pathlib.Path:
     """Validate path."""
-    try:
-        get_path = pathlib.Path(path)
-    except OSError as error:
-        _LOGGER.error("Can not access file %s, error %s", path, error)
-        raise NotValidError from error
+    get_path = pathlib.Path(path)
+    exist = get_path.exists()
+    is_file = get_path.is_file()
+    if not exist or not is_file:
+        _LOGGER.error("Can not access file %s", path)
+        raise NotValidError
 
     if not hass.config.is_allowed_path(path):
-        _LOGGER.error("Filepath %s is not valid or allowed", path)
+        _LOGGER.error("Filepath %s is not allowed", path)
         raise NotAllowedError
 
     return get_path
```

Comment:
We may want to move the call to `Path.absolute` here. It seems to not be a `PurePath` method so may do I/O. It's not documented but it's there in the source. They've commented that it's not tested, so maybe that's why they haven't documented it yet?

Reply:
Should we use resolve() instead?
` Make the path absolute, resolving all symlinks on the way and also normalizing it (for example turning slashes into backslashes under Windows).`

Reply:
I see now `Path.absolute` is used by `Path.resolve` so that would not help anything.
Can move it instead